src/ConnectedMode/ServerSentEvents/IServerSentEventSourcePublisher.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents.TEMP;
+
+namespace SonarLint.VisualStudio.ConnectedMode.ServerSentEvents
+{
+    internal interface IServerSentEventSourcePublisher<T> where T : IServerSentEvent
+    {
+        Task Publish(T serverEvent);
```

Comment:
doc comment what it returns

Reply:
Describe why it's a Task and not a void?

Reply:
Yes (should probably be PublishAsync). i.e. can it throw?

Reply:
done